tsl/test/expected/compression_defaults.out

```diff
@@ -0,0 +1,205 @@
+-- This file and its contents are licensed under the Timescale License.
+-- Please see the included NOTICE for copyright information and
+-- LICENSE-TIMESCALE for a copy of the license.
+\c :TEST_DBNAME :ROLE_SUPERUSER
+-- statitics on
+CREATE TABLE "public"."metrics" (
+    "time" timestamp with time zone NOT NULL,
+    "device_id" "text",
+    "val" double precision
+) WITH (autovacuum_enabled=0);
+SELECT create_hypertable('public.metrics', 'time', create_default_indexes=>false);
+  create_hypertable   
+----------------------
+ (1,public,metrics,t)
+(1 row)
+
+insert into metrics SELECT t, 1, extract(epoch from t) from generate_series
+        ( '2007-02-01'::timestamp
+        , '2008-04-01'::timestamp
+        , '1 day'::interval) t;
+insert into metrics SELECT t, 2, extract(epoch from t) from generate_series
+        ( '2009-02-01'::timestamp
+        , '2010-04-01'::timestamp
+        , '1 day'::interval) t;
+ANALYZE metrics;
+--use the best-scenario unique index
+CREATE UNIQUE INDEX test_idx ON metrics(device_id, time);
+SELECT _timescaledb_functions.get_segmentby_defaults('public.metrics');
+            get_segmentby_defaults            
+----------------------------------------------
+ {"columns": ["device_id"], "confidence": 10}
+(1 row)
+
+SELECT _timescaledb_functions.get_orderby_defaults('public.metrics', ARRAY['device_id']);
+NOTICE:  orderby names: {time} <NULL>
+            get_orderby_defaults             
+---------------------------------------------
+ {"clauses": ["time DESC"], "confidence": 8}
+(1 row)
+
+--use a high-cardinalit column in the index
+drop index test_idx;
+CREATE UNIQUE INDEX test_idx ON metrics(val, time);
+SELECT _timescaledb_functions.get_segmentby_defaults('public.metrics');
+                                                                              get_segmentby_defaults                                                                              
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+ {"columns": [], "message": "Several columns are potential segment by candidates. Please use the segment_by option to explicitly specify the segment_by column", "confidence": 0}
```

Comment:
This message does not seem to be correct: there are no candidate columns.

Reply:
Changed message